From 85a5a5bc062a39ba14425097a536f70cc78022d0 Mon Sep 17 00:00:00 2001
From: hmtheboy154 <buingoc67@gmail.com>
Date: Thu, 24 Apr 2025 10:49:47 +0000
Subject: [PATCH 22/42] init.sh: Allow delaying init_hal_sensors()

There are cases that Android itself boot up too fast that iio sensors
drivers just isn't ready yet. Add a cmdline to delay init_hal_sensors so
that the iio sensors drivers can be loaded first.

Signed-off-by: hmtheboy154 <buingoc67@gmail.com>
---
 init.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/init.sh b/init.sh
index 94768c3..47bcf41 100644
--- a/init.sh
+++ b/init.sh
@@ -697,6 +697,11 @@ function init_hal_thermal()
 
 function init_hal_sensors()
 {
+	# Pause for amount of seconds before initializing sensors
+	if [ -n "$SENSORS_DELAY_INIT" ]; then
+		sleep "$SENSORS_DELAY_INIT"
+	fi
+
     if [ "$SENSORS_FORCE_KBDSENSOR" == "1" ]; then
         # Option to force kbd sensor
         hal_sensors=kbd
-- 
2.43.0

